Sources/SwiftLexicalLookup/ScopeSyntax.swift

```diff
@@ -0,0 +1,109 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension SyntaxProtocol {
+  /// Returns all names that `for` refers to at this syntax node.
+  /// Optional configuration can be passed as `with` to customize the lookup behavior.
```

Comment:
We generally use the internal parameter names to refer to the parameters. This matches docc’s behavior and allows us to disambiguate which parameter we mean if we have a function like `func (with x: Int, with: y)` or allow us to refer to parameters at all if they are unnamed.

```suggestion
  /// Returns all names that `name` refers to at this syntax node.
  /// Optional configuration can be passed as `config` to customize the lookup behavior.
```

Similar in the rest of the comment.

Reply:
Thank you for brining this up. I was thinking from the clients' perspective, but using internal names indeed makes more sense. I hope I fixed it in all relevant places.

Reply:
I initially also thought that the external names made more sense and had a whole conversation about it that convinced me otherwise 😉